tracing-mock/src/ancestry.rs

```diff
@@ -0,0 +1,145 @@
+//! Define the ancestry of an event or span.
+//!
+//! See the documentation on the [`Ancestry`] enum for further details.
+
+use tracing_core::{
+    span::{self, Attributes},
+    Event,
+};
+
+/// The ancestry of an event or span.
+///
+/// An event or span can have an explicitly assigned parent, or be an explicit root. Otherwise,
+/// an event or span may have a contextually assigned parent or in the final case will be a
+/// contextual root.
+#[derive(Debug, Eq, PartialEq)]
+pub enum Ancestry {
+    /// The event or span has an explicitly assigned parent (created with `parent: span_id`) with
+    /// the specified name.
+    HasExplicitParent(String),
+    /// The event or span is an explicitly defined root. It was created with `parent: None` and
+    /// has no parent.
+    IsExplicitRoot,
+    /// The event or span has a contextually assigned parent with the specified name. Additionally,
+    /// it has no explicitly assigned parent.
```

Comment:
I'm not parsing this sentence right: are you trying to say it has no explicitly defined root..?

Reply:
Here I want to say that the span or event has no explicitly defined parent (which means the `parent:` field in a span or event macro was **not** used). Instead, it has a parent that was contextually assigned.

Does that make sense? I see how it's not super clear, I can have another go at rewriting it if it's still not making sense.